src/data_access/UserDataAccessObject.java

```
@@ -47,4 +50,15 @@ private void save() {
         // TODO Implement this method
     }
 
+    @Override
+    public Playlist getPlaylist(String playlistName, String username) {
+        // Return the Playlist with the given name belonging to the given user
+        User user = users.get(username);
+        for (Playlist playlist: user.getPlaylists()) {
+            if (playlist.getName().equals(playlistName)) {
+                return playlist;
+            }
+        }
+        return null; // This line should never be reached as the playlist name will be valid
```

Instead of returning null, you might want to throw an exception (which will be considered as a better practice)